src/Adapter/Employee/NavigationHandler.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Employee;
+
+use PrestaShop\PrestaShop\Adapter\LegacyContext;
+use PrestaShop\PrestaShop\Core\Employee\NavigationHandlerInterface;
+
+/**
+ * Class NavigationHandler handles collapsing/expanding the navigation for context employee.
+ */
+final class NavigationHandler implements NavigationHandlerInterface
+{
+    /**
+     * @var LegacyContext
+     */
+    private $legacyContext;
+
+    /**
+     * @param LegacyContext $legacyContext
+     */
+    public function __construct(LegacyContext $legacyContext)
+    {
+        $this->legacyContext = $legacyContext;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function toggleNavigation($shouldCollapse)
```

Comment:
Same here: in a classic application, we would never modify cookies in Core/Adapter code. We would perform this at Controller level because cookies are a web-only system and Core/Adapter code should be agnostic (it should be able to run for both CLI apps and web apps).

So I think it would be good to highlight the fact cookies are being modified here, else developers might not expect that
```suggestion
    public function togglesNavigationMenuInCookies($shouldCollapse)
```